src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/FileBasedPrograms/FileBasedProgramsProjectSystem.cs

```diff
@@ -117,6 +117,13 @@ public async ValueTask TryRemoveMiscellaneousDocumentAsync(DocumentUri uri)
     protected override async Task<RemoteProjectLoadResult?> TryLoadProjectInMSBuildHostAsync(
         BuildHostProcessManager buildHostProcessManager, string documentPath, CancellationToken cancellationToken)
     {
+        if (Path.GetExtension(documentPath) != ".cs")
```

Comment:
This check can/should be up on line 76 where we even consider whether we want to launch this. @RikkiGibson want to make a full fix here?

Reply:
And now that my work to get the miscellaneous files unit tests running on this implementation is done, this should be testable. It might be a small fix to get the existing tests catching this problem.

Reply:
I got a test okay, just couldn't get it to fail. I'm missing whatever ends up calling try apply changes I think. Maybe a rebuild of the project or something?

@RikkiGibson let know if you want me to push up what I have, in case it help.

Reply:
Go ahead, I can pull it down and triage.